static/app/components/integrationExternalMappingForm.tsx

```diff
@@ -4,127 +4,153 @@ import capitalize from 'lodash/capitalize';
+              ...sentryNamesMapper(result),
+            ]);
           }
-          this.props.onResults?.(result);
           return optionMapper(sentryNamesMapper(result));
         },
-      });
-    }
-    if (type === 'team') {
-      fields.push({
-        name: 'teamId',
-        type: 'select_async',
+      },
+    ];
+    // We only add the field for externalName if it's the full (not inline) form
+    if (!isInline) {
+      fields.unshift({
+        name: 'externalName',
+        type: 'string',
         required: true,
-        label: tct('Sentry [type]', {type: capitalize(type)}),
-        placeholder: t(`Choose your Sentry Team`),
-        url,
-        onResults: result => {
-          // For organizations with >100 teams, we want to make sure their
-          // saved mapping gets populated in the results if it wouldn't have
-          // been in the initial 100 API results, which is why we add it here
-          if (mapping && !result.find(({id}) => id === mapping.teamId)) {
-            result = [{id: mapping.teamId, name: mapping.sentryName}, ...result];
-          }
-          // The team needs `this.props.onResults` so that we have team slug
-          // when a user submits a team mapping, the endpoint needs the slug
-          // as a path param: /teams/${organization.slug}/${team.slug}/external-teams/
-          this.props.onResults?.(result);
-          return optionMapper(sentryNamesMapper(result));
-        },
+        label: isInline ? undefined : tct('External [type]', {type: capitalize(type)}),
+        placeholder: t(`${type === 'user' ? '@username' : '@org/teamname'}`),
       });
     }
     return fields;
   }
 
-  render() {
-    const {onSubmitSuccess, onCancel, mapping, baseEndpoint, onSubmit} = this.props;
+  get extraFormFieldProps() {
+    const {isInline, mapping, type} = this.props;
+    return isInline
+      ? {
+          // We need to submit the entire model since it could be a new one or an update
+          getData: () => this.model.getData(),
+          // We need to update the model onBlur for inline forms since the model's 'onPreSubmit' hook
+          // does NOT run when using `saveOnBlur`.
+          onBlur: value => {
+            if (!mapping) {
+              return;
+            }
+            const updatedMapping = {
+              ...mapping,
+              [`${type}Id`]: value,
+            };
+            this.updateModelFromMapping(updatedMapping);
+          },
+        }
+      : {flexibleControlStateSize: true};
+  }
 
-    // endpoint changes if we are making a new row or updating an existing one
-    const endpoint = !baseEndpoint
-      ? undefined
-      : mapping
-      ? `${baseEndpoint}${mapping.id}/`
-      : baseEndpoint;
-    const apiMethod = !baseEndpoint ? undefined : mapping ? 'PUT' : 'POST';
+  // This function is necessary since the endpoint we submit to changes depending on the value selected
+  updateModelFromMapping(mapping?: ExternalActorMapping) {
```

Comment:
This is where we update the `apiMethod` and `apiEndpoint` get updated, for the reasons I mentionned above.